Content.Shared/Pinpointer/SharedPinpointerSystem.cs

```diff
@@ -20,6 +31,33 @@ private void GetCompState(EntityUid uid, PinpointerComponent pinpointer, ref Com
             };
         }
 
+        /// <summary>
+        ///     Set the target if capable
+        /// </summary>
+        private void OnAfterInteract(EntityUid uid, PinpointerComponent component, AfterInteractEvent args)
+        {
+            if (!args.CanReach || args.Target is not { } target)
+                return;
+
+            if (!component.CanRetarget || component.IsActive)
+                return;
+
+            // TODO add doafter once the freeze is lifted
+            args.Handled = true;
+            component.Target = args.Target;
+            _adminLogger.Add(LogType.Action, LogImpact.Low, $"{ToPrettyString(args.User):player} set target of {ToPrettyString(uid):pinpointer} to {ToPrettyString(component.Target.Value):target}");
+            if (component.UpdateTargetName)
+                component.TargetName = component.Target == null ? null : CompOrNull<MetaDataComponent>(component.Target.Value)?.EntityName;
+        }
+
+        private void OnExamined(EntityUid uid, PinpointerComponent component, ExaminedEvent args)
+        {
+            if (!args.IsInDetailsRange || component.TargetName == null)
+                return;
+
+            args.PushMarkup(Loc.GetString("examine-pinpointer-linked", ("target", component.TargetName)));
```

Comment:
make it say "It is tracking ???" or "It is tracking XXX" or smt like that if the target is null or UpdateTargetName is false

Reply:
that is fine if UpdateTargetName is true by default in the nuke disk pinpointer, but it should update to ??? when set to false if the code is working correctly